prqlc/prql-compiler/src/sql/dialect.rs

```diff
@@ -187,7 +202,11 @@ impl dyn DialectHandler {
     }
 }
 
-impl DialectHandler for GenericDialect {}
+impl DialectHandler for GenericDialect {
+    fn translate_chrono_item(&self, _item: Item) -> Result<String> {
+        bail!("Date formatting requires a dialect")
```

Comment:
I would guess that the `%Y`-like strings are the most common. I would probably vote to default to that for the generic.

Though this is one of the first times where the ecosystem is fairly evenly split, so no strong view

(And doesn't need to be in this PR)

Reply:
I don't know to be honest
- BigQuery is close to MySQL for the common ones (%Y, %d, %m, %H, %M, %S so why not but not for the full weekday for example which is common)
- Clickhouse supports both percentage (MySQL-like) and Joda syntax (MSSQL-like). `%Y` is understood but for example `%m` and `%M` are for months (Urghhh I know...)
- DuckDB is very close to `chrono` syntax so ok
- GlareDB doesn't support it yet I guess. I digged into datafusion doc and asked on discord without any response
- MySQL yes it works
- MSSQL is `yyyy`
- Postgres (and hence Snowflake) is `YYYY`

I think it's dangerous to have defaults that are for only 1/3 of the supported DBs (and even for them we'll have issues for others)
This is why I chose this approach

Reply:
Yeah, that is even more balkanized than I thought before...

Reply:
> balkanized

Haha, first time I seen this term, but it describes the state here nicely :D

----

I'd say that for the generic dialect, we avoid confusing introducing a combination of the formats of other databases and just use strftime format (i.e. don't translate), since generic dialect is not meant to work at all, but it should be something that is displayed in the docs.